index.js

```diff
@@ -0,0 +1,39 @@
+const validateLineup = (lineup) => {
+  function getSalaries(lineup) {
+    return lineup.salary
+  }
+  function getSalarySum(acc, n) {
+    return acc + n
+  }
+  function getTeamId(lineup) {
+    return lineup.teamId
+  }
+  function getGameId(lineup) {
+    return lineup.gameId
+  }
+  function getPositions(lineup) {
+    return lineup.position
+  }
+
+  const salaryList = lineup.map(getSalaries)
+  const salarySum = salaryList.reduce(getSalarySum, 0)
```

Comment:
This could be done in a single reduce, instead you're looping twice as much as you need to.